packages/flutter/lib/src/cupertino/segmented_control.dart

```diff
@@ -0,0 +1,683 @@
+            color: (_selectedValue == childKey)
+                ? CupertinoColors.white
+                : CupertinoColors.activeBlue,
+          );
+      final IconThemeData iconTheme = IconThemeData(
+        color: (_selectedValue == childKey)
+            ? CupertinoColors.white
+            : CupertinoColors.activeBlue,
+      );
+
+      Widget child = widget.children.elementAt(i);
+      child = GestureDetector(
+        onTapDown: (TapDownDetails event) {
+          _onTapDown(childKey);
+        },
+        onTapUp: (TapUpDetails event) {
+          _onTapUp();
+        },
+        onTapCancel: () {
+          _onTapCancel();
+        },
+        onTap: () {
+          _onTap(widget.children.elementAt(i).key, childKey);
+        },
+        child: new IconTheme(
+            data: iconTheme,
+            child: DefaultTextStyle(
+              style: textStyle,
+              overflow: TextOverflow.ellipsis,
+              child: Semantics(
+                  inMutuallyExclusiveGroup: true,
+                  selected: _selectedValue != null
+                      ? _selectedValue.value == i
+                      : false,
+                  child: child),
+            )),
+      );
+      _gestureChildren.add(child);
+    }
+
+    final Widget box = new _SegmentedControlRenderWidget(
+      children: _gestureChildren,
+      selectedValue: _selectedValue,
+      pressedValue: _pressedValue,
+    );
+
+    return Padding(
+      padding: _kHorizontalItemPadding.resolve(Directionality.of(context)),
+      child: UnconstrainedBox(
+        constrainedAxis: Axis.horizontal,
+        child: box,
+      ),
+    );
+  }
+}
+
+class _SegmentedControlRenderWidget extends RenderObjectWidget {
+  const _SegmentedControlRenderWidget({
+    Key key,
+    @required this.children,
+    @required this.selectedValue,
+    @required this.pressedValue,
+  });
+
+  final List<Widget> children;
+  final ValueKey<int> selectedValue;
+  final ValueKey<int> pressedValue;
+
+  @override
+  RenderObjectElement createElement() => new _RenderSegmentedControlElement(this);
+
+  @override
+  RenderObject createRenderObject(BuildContext context) {
+    return new _RenderSegmentedControl(
+      textDirection: Directionality.of(context),
+      children: children,
+      selectedValue: selectedValue,
+      pressedValue: pressedValue,
+    );
+  }
+
+  @override
+  void updateRenderObject(
+      BuildContext context, _RenderSegmentedControl renderObject) {
+    renderObject.markNeedsPaint();
+    renderObject
+      ..textDirection = Directionality.of(context)
+      ..children = children
+      ..selectedValue = selectedValue
+      ..pressedValue = pressedValue;
+  }
+}
+
+class _RenderSegmentedControlElement extends RenderObjectElement {
+  _RenderSegmentedControlElement(_SegmentedControlRenderWidget segmentedControl)
+      : super(segmentedControl);
+
+  final List<Element> _slots = <Element>[];
```

Comment:
if you just have a list of children, just use the existing handling for that. (MultiChildRenderObjectWidget and so on). Same on the RenderBox side; we have lots of existing code to handle multiple children, just reuse that.